### PR TITLE
[CR] [WIP] Change handling of recoil penalty

### PR DESCRIPTION
**Pull request by mugling**
Inspired by a forum post (needful things) and is dependent upon #14627.

Applies a penalty to recoil after each successful dodge which is scaled according to the volume of the wielded gun and offset by increased skill in dodging. Pistols are therefore far more desirable in close quarters although for later game characters SMGs also become viable. Would work well if I was to implement (separate PR) shortened barrels reducing volume.

I'd like to further extend upon this with the goal of adding more variety to the firearms mechanics so marked as `[CR]` `[WIP]` for now.

### CODE REVIEW

File: src/monster.cpp [hunk 1]
```diff
@@ -924,6 +923,9 @@ void monster::melee_attack(Creature &target, bool, const matec_id&) {
                             target.disp_name(true).c_str());
             }
         }
+        if( !is_hallucination() ) {
```
Comment (kevingranade) [inline src/monster.cpp]: hrm, the enemy isn't real, but you actually dodged, it seems like it should trigger anyway. (not directly related to your PR, so I'm not asking for a fix)

**Comment by kevingranade** [comment]
I'm merge testing this right now, it looks like it's uncovered a bug where you can't dodge when aiming.
I don't remember disabling dodging while aiming anyway...

File: src/monster.cpp (hunk 1 above)
Comment (mugling) [inline src/monster.cpp]: Sure, this could potentially go in #14636 once we determine how to proceed?

**Comment by mugling** [comment]
See @Coolthulhu's comments in #14627 - ce3386b could make it hard for characters with low dodging to ever improve?

**Pull request merged**